src/NHibernate/Linq/Visitors/ResultOperatorProcessors/ProcessFetch.cs

```diff
@@ -68,8 +68,18 @@ private void Process(FetchRequestBase resultOperator, QueryModelVisitor queryMod
 				}
 			}
 
-			var alias = queryModelVisitor.Model.GetNewName("_");
-			tree.AddFromClause(tree.TreeBuilder.LeftFetchJoin(memberPath, tree.TreeBuilder.Alias(alias)));
+			var relatedJoin = queryModelVisitor.RelatedJoinFetchRequests.FirstOrDefault(o => o.Value == resultOperator).Key;
```

Comment:
Shouldn't this logic be moved to  `if (resultOperator is FetchOneRequest)` block? It seems we shouldn't try to reuse collection fetches

Reply:
It should, moved.